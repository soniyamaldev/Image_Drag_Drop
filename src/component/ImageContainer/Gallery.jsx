import React, {Component} from 'react';
import {SortableContainer, SortableElement, arrayMove} from 'react-sortable-hoc';
import { confirmAlert } from 'react-confirm-alert'; // Import
import 'react-confirm-alert/src/react-confirm-alert.css'; // Import css
import {imageArray} from '../../constants/config';
import '../../App.css'

const ImageWrapper = SortableContainer(({imageArray}) => {
    return (
        <div>
            <h2>Pictures Zone (Drag and Drop)</h2>
            {imageArray.map((value, index) => (
                <Images key={index} index={index} value={value} />
            ))}
        </div>
    );
});

const Images = SortableElement(({value}) => <div className="img">
    <center>
    <img src={value} className={'img-container'}/>
    </center>
</div>);

export default class Gallery extends Component {
    state = {
        imageArray: imageArray
    };

    onSortEnd = ({oldIndex, newIndex}) => {
        if(oldIndex !== newIndex){
            confirmAlert({
                title: '',
                message: `Do you want to replace image${oldIndex + 1} for image${newIndex + 1} ?`,
                buttons: [
                    {
                        label: 'Yes',
                        onClick: () => {
                            this.setState(({imageArray}) => ({
                                imageArray: arrayMove(imageArray, oldIndex, newIndex),
                            }));
                        }
                    },
                    {
                        label: 'No',
                        onClick: () => console.log("No movement")
                    }
                ]
            });
        }
    };

    render() {
        return <ImageWrapper imageArray={this.state.imageArray} onSortEnd={this.onSortEnd} />;
    }
}
