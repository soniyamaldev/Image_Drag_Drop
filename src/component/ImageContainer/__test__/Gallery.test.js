import React from "react";
import {render} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Gallery from "../Gallery";

describe("Banner Component", () => {
    it("should match snapshot when properties are present", () => {
        const {container} = render(
            <Gallery/>
        );

        expect(container).toMatchSnapshot();
    });
});
