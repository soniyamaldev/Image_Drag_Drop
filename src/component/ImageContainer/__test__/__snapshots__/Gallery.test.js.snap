// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component should match snapshot when properties are present 1`] = `
<div>
  <div>
    <h2>
      Pictures Zone (Drag and Drop)
    </h2>
    <div
      class="img"
    >
      <center>
        <img
          class="img-container"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRLD3hlhK6Ktd7odrFRPYK9c_jDY2Xx0wkfsgp7Xmzdz07bjcck&usqp=CAU"
        />
      </center>
    </div>
    <div
      class="img"
    >
      <center>
        <img
          class="img-container"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQ52tkfcdyn7u2Hvip9B8ucKkMl8TXugr2wOU_mglrZUPGXmCJ1&usqp=CAU"
        />
      </center>
    </div>
    <div
      class="img"
    >
      <center>
        <img
          class="img-container"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTtFt0ZPPjvZPGciDG42NuoXCb9YyQXx1mEZ17Yq1tws151MqBz&usqp=CAU"
        />
      </center>
    </div>
    <div
      class="img"
    >
      <center>
        <img
          class="img-container"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcTMA_QPKGS5LigXQx9Cw5eED5QaXlKvjLcYRp5terwKjQCPRmcg&usqp=CAU"
        />
      </center>
    </div>
    <div
      class="img"
    >
      <center>
        <img
          class="img-container"
          src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcSOvl5Flx9-qIflBN3pkyHfDk0x70el55lCZy2KNFqpkHGeiLp0&usqp=CAU"
        />
      </center>
    </div>
  </div>
</div>
`;
